main/src/ca/uwaterloo/flix/language/phase/Kinder.scala

```diff
@@ -71,18 +70,19 @@ object Kinder {
 
     val restrictableEnums = ParOps.parMapValues(root.restrictableEnums)(visitRestrictableEnum(_, taenv, root))
 
-    val traitsVal = visitTraits(root, taenv, oldRoot, changeSet)
+    val traits = visitTraits(root, taenv, oldRoot, changeSet)
 
-    val defsVal = visitDefs(root, taenv, oldRoot, changeSet)
+    val defs = visitDefs(root, taenv, oldRoot, changeSet)
 
-    val instancesVal = ParOps.parTraverseValues(root.instances)(traverse(_)(i => visitInstance(i, taenv, root)))
+    // Should `instances` not be root.instances.map(ParOps.parMap)?
+    // I.e., for every trait, we check all instances in parallel so we get a parallel task for every instance
+    // instead of check all traits in parallel which spawns k threads each does synchronous code
+    val instances = ParOps.parMapValues(root.instances)(_.map(visitInstance(_, taenv, root)))
```

Comment:
@magnus-madsen see comment above

Reply:
Not sure. What does @mlutze think?

Reply:
Ill merge this now, but generally avoid comments in code-- better to put them in the PR :)

Reply:
By comments I mean questions.

We do want comments in the code 🤡

Reply:
Hehe yeah, I was planning to remove that, though... but too late :o

Reply:
I can benchmark both versions on the server to check if has an impact

Reply:
You can make a follow up-PR with your proposed change